aries_vcx/src/handlers/out_of_band/sender/sender.rs

```diff
@@ -0,0 +1,84 @@
+use crate::handlers::out_of_band::{OutOfBand, GoalCode, HandshakeProtocol};
+use crate::messages::attachment::{AttachmentId, AttachmentEncoding};
+use crate::messages::connection::service::ServiceResolvable;
+use crate::messages::a2a::A2AMessage;
+use crate::messages::a2a::message_type::MessageType;
+use crate::messages::a2a::message_family::MessageFamilies;
+use crate::error::prelude::*;
+
+impl OutOfBand {
+    pub fn create() -> Self {
+        Self::default()
+    }
+
+    pub fn set_label(mut self, label: &str) -> Self {
+        self.label = Some(label.to_string());
+        self
+    }
+
+    pub fn set_goal_code(mut self, goal_code: GoalCode) -> Self {
+        self.goal_code = Some(goal_code);
+        self
+    }
+
+    pub fn set_goal(mut self, goal: &str) -> Self {
+        self.goal = Some(goal.to_string());
+        self
+    }
+
+    pub fn append_service(mut self, service: ServiceResolvable) -> Self {
+        self.services.push(service);
+        self
+    }
+
+    pub fn append_handshake_protocol(mut self, protocol: HandshakeProtocol) -> VcxResult<Self> {
+        let new_protocol = match protocol {
+            HandshakeProtocol::ConnectionV1 => MessageType::build(MessageFamilies::Connections, ""),
+            HandshakeProtocol::DidExchangeV1 => { return Err(VcxError::from(VcxErrorKind::ActionNotSupported)) }
```

Comment:
Good reminder of protocol work ahead of us :-) Went ahead and created an issue to track this
https://github.com/hyperledger/aries-vcx/issues/346